contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,312 @@
+    /// withdrawal of the validators. It is strictly concerned with WETH that has been deposited and is waiting to
+    /// be staked.
+    uint256 public depositedWethAccountedFor;
+
+    // For future use
+    uint256[49] private __gap;
+
+    /// @param _baseConfig Base strategy config with platformAddress (ERC-4626 Vault contract), eg sfrxETH or sDAI,
+    /// and vaultAddress (OToken Vault contract), eg VaultProxy or OETHVaultProxy
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _ssvToken Address of the Erc20 SSV Token contract
+    /// @param _ssvNetwork Address of the SSV Network contract
+    /// @param _maxValidators Maximum number of validators that can be registered in the strategy
+    /// @param _feeAccumulator Address of the fee accumulator receiving execution layer validator rewards
+    /// @param _beaconChainDepositContract Address of the beacon chain deposit contract
+    constructor(
+        BaseStrategyConfig memory _baseConfig,
+        address _wethAddress,
+        address _ssvToken,
+        address _ssvNetwork,
+        uint256 _maxValidators,
+        address _feeAccumulator,
+        address _beaconChainDepositContract
+    )
+        InitializableAbstractStrategy(_baseConfig)
+        ValidatorAccountant(
+            _wethAddress,
+            _baseConfig.vaultAddress,
+            _beaconChainDepositContract,
+            _ssvNetwork,
+            _maxValidators
+        )
+    {
+        SSV_TOKEN = _ssvToken;
+        FEE_ACCUMULATOR_ADDRESS = payable(_feeAccumulator);
+    }
+
+    /// @notice initialize function, to set up initial internal state
+    /// @param _rewardTokenAddresses Address of reward token for platform
+    /// @param _assets Addresses of initial supported assets
+    /// @param _pTokens Platform Token corresponding addresses
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+    }
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just checks the asset is WETH and emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    /// @param _asset Address of asset to deposit. Has to be WETH.
+    /// @param _amount Amount of assets that were transferred to the strategy by the vault.
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_asset == WETH, "Unsupported asset");
+        depositedWethAccountedFor += _amount;
```

Comment:
Withdrawing does not adjust `depositedWethAccountedFor`. If the strategist were to

- deposit(32 weth)
- withdraw(32 weth)
- deposit(32 weth)

then the `depositedWethAccountedFor` would be doubled. This would also cause a subsequent `depositAll()` to revert when doing `wethBalance - depositedWethAccountedFor;`

Reply:
Good pickup. We should have caught this.

I've put this change in a separate PR back to this branch as its non-trivial https://github.com/OriginProtocol/origin-dollar/pull/2091